test/test-generate-request-details.js

```diff
@@ -243,4 +244,26 @@ suite('Test Generate Request Details', function() {
     assert.equal(details.headers.Topic, extraOptions.headers.Topic);
     assert.equal(details.headers.Urgency, extraOptions.headers.Urgency);
   });
+
+  test('Audience contains port', function() {
+    const subscription = {
+      endpoint: 'http://example.com:4242/life-universe-and-everything'
+    };
+
+    const extraOptions = {
+      vapidDetails: {
+        subject: 'mailto:example@example.com',
+        publicKey: vapidKeys.publicKey,
+        privateKey: vapidKeys.privateKey
+      }
+    };
+
+    const message = undefined;
+    const requestDetails = webPush.generateRequestDetails(subscription, message, extraOptions);
+    const authHeader = requestDetails.headers.Authorization;
+    const audience = jws.decode(authHeader.slice(8)).payload.aud;
+
+    assert.ok(audience, 'Audience does not exist');
```

Comment:
For this and the following, please write what you are asserting instead of what makes the assertion fail (`Audience exists` instead of `Audience does not exist`).